packages/web/src/shared/components/form/RichTextInput.tsx

```diff
@@ -71,6 +75,10 @@ const RichTextInput = forwardRef<RichTextInputRef, RichTextInputProps>(
       });
     }, [value, editor]);
 
+    useEffect(() => {
+      if (autoFocus) editor?.commands.focus();
+    }, [autoFocus, editor]);
```

Comment:
We should use `useImperativeHandle` to implement the `focus` function of the ref. Then you don't need all of the hoops to keep track of the focus state in the sidebar. Just grab a ref to the note input and then execute focus in the imperative handle hook

Reply:
I'm having trouble getting this to work consistently. Could you review my code to see what I am doing wrong?

Reply:
Looks like the issue is that there is a race condition between the fall to focus, and the creation of the editor state. This means that when focus is called, if the rich text editor has not mounted yet, the editor will be null for long enough that focus doesn't do anything. We can fix this by adding `unmount={false}` to each `Tab.Panel` in the sidebar so that the tabs are always mounted.